main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/bwplotka/mdox/pkg/clilog"
+	extflag "github.com/efficientgo/tools/extkingpin"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/observatorium/obsctl/pkg/extkingpin"
+	"github.com/observatorium/obsctl/pkg/version"
+	"github.com/oklog/run"
+	"github.com/pkg/errors"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+const (
+	logFormatLogfmt = "logfmt"
+	logFormatJson   = "json"
+	logFormatCLILog = "clilog"
+)
+
+func setupLogger(logLevel, logFormat string) log.Logger {
+	var lvl level.Option
+	switch logLevel {
+	case "error":
+		lvl = level.AllowError()
+	case "warn":
+		lvl = level.AllowWarn()
+	case "info":
+		lvl = level.AllowInfo()
+	case "debug":
+		lvl = level.AllowDebug()
+	default:
+		panic("unexpected log level")
+	}
+	switch logFormat {
+	case logFormatJson:
+		return level.NewFilter(log.NewJSONLogger(log.NewSyncWriter(os.Stderr)), lvl)
+	case logFormatLogfmt:
+		return level.NewFilter(log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr)), lvl)
+	case logFormatCLILog:
+		fallthrough
+	default:
+		return level.NewFilter(clilog.New(log.NewSyncWriter(os.Stderr)), lvl)
+	}
+}
+
+func main() {
+	app := extkingpin.NewApp(kingpin.New(filepath.Base(os.Args[0]), `obsctl`).Version(version.Version))
+	logLevel := app.Flag("log.level", "Log filtering level.").Default("info").Enum("error", "warn", "info", "debug")
+	logFormat := app.Flag("log.format", "Log format to use.").Default(logFormatCLILog).Enum(logFormatLogfmt, logFormatJson, logFormatCLILog)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	// Auth commands.
+	registerLogin(ctx, app)
+	registerLogout(ctx, app)
+	registerCurrent(ctx, app)
+	registerSwitch(ctx, app)
+
+	// Metrics operations.
+	// TODO(saswatamcode): Scope these under metrics flag, to support logs and traces in future.
+	registerRead(ctx, app)
+	registerRules(ctx, app)
+	registerQuery(ctx, app)
+
+	cmd, runner := app.Parse()
+	logger := setupLogger(*logLevel, *logFormat)
+
+	var g run.Group
+	g.Add(func() error {
+		return runner(ctx, logger)
+	}, func(err error) {
+		cancel()
+	})
+
+	// Listen for termination signals.
+	{
+		cancel := make(chan struct{})
+		g.Add(func() error {
+			return interrupt(logger, cancel)
+		}, func(error) {
+			close(cancel)
+		})
+	}
+
+	if err := g.Run(); err != nil {
+		if *logLevel == "debug" {
+			// Use %+v for github.com/pkg/errors error to print with stack.
+			level.Error(logger).Log("err", fmt.Sprintf("%+v", errors.Wrapf(err, "%s command failed", cmd)))
+			os.Exit(1)
+		}
+		level.Error(logger).Log("err", errors.Wrapf(err, "%s command failed", cmd))
```

Comment:
We can wrap errors with stdlib now, so I am not too sold on the idea of using `github.com/pkg/errors`. 
```suggestion
		level.Error(logger).Log("err", fmt.Errorf("%s command failed: %w", cmd, err))
```